pkg/vsphere/tasks/waiter.go

```diff
@@ -89,6 +102,56 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 	}
 }
 
+func fixTask(ctx context.Context, sess *session.Session, target object.Reference) error {
+	switch vmm := target.(type) {
+	case *object.VirtualMachine:
+		vm := vm.NewVirtualMachine(ctx, sess, target.Reference())
```

Comment:
You should get rid of the `sess` parameter to `NewVirtualMachine`. Its leaking badly into this API, and the `vm.VirtualMachine` class really uses it in only two places -- both of which can be replaced with a call to [`Client()`](https://github.com/vmware/vic/blob/master/vendor/github.com/vmware/govmomi/object/common.go#L62).